barretenberg/cpp/src/barretenberg/goblin/goblin.hpp

```diff
@@ -279,6 +209,19 @@ class GoblinVerifier {
         , translator_verification_key(input.translator_verification_key)
     {}
 
+    /**
+     * @brief Append recursive verification of a merge proof to a provided circuit
+     *
+     * @param circuit_builder
+     * @return PairingPoints
+     */
+    static PairingPoints recursive_verify_merge(Builder& circuit_builder, const StdlibProof<Builder>& proof)
```

Comment:
Just moving this logic from the prover to the verifier. Its debatable whether it should even live here or if CICV should just interact with the Merge classes directly. Keeping this model for now since it plausibly makes sense for Goblin to encapsulate the merge logic.